presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimate.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cost;
+
+import com.facebook.presto.sql.planner.Symbol;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.lang.Double.NaN;
+import static java.lang.Double.isNaN;
+
+public class PlanNodeStatsEstimate
+{
+    public static final PlanNodeStatsEstimate UNKNOWN_STATS = PlanNodeStatsEstimate.builder().build();
+    public static final double DEFAULT_DATA_SIZE_PER_COLUMN = 10;
+
+    private final double outputRowCount;
+    private final Map<Symbol, SymbolStatsEstimate> symbolStatistics;
+
+    private PlanNodeStatsEstimate(double outputRowCount, Map<Symbol, SymbolStatsEstimate> symbolStatistics)
+    {
+        this.outputRowCount = outputRowCount;
+        this.symbolStatistics = ImmutableMap.copyOf(symbolStatistics);
+    }
+
+    public double getOutputRowCount()
+    {
+        return outputRowCount;
+    }
+
+    public double getOutputSizeInBytes()
```

Comment:
also, maybe if there are no symbol stats, but there is a row count, this should estimate some value based on row count.  Otherwise incomplete stats end up being useless